### PR TITLE
[New] Show line of sight between points

### PR DESCRIPTION
**Pull request by CalebRas**
## Description

This PR implements `Show line of sight between points` in `Analysis` category.
URL to README: https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Caleb/New-ShowLineOfSightBetweenPoints/Shared/Samples/Show%20line%20of%20sight%20between%20points

## Linked Issue(s)

- `swift/issues/4150`

## How To Test

* Tap on the map to set the observer location.
* Tap and hold on the map to set the target location.

## Screenshots

show-line-of-sight-between-points

### CODE REVIEW

File: Shared/Samples/Show line of sight between points/ShowLineOfSightBetweenPointsView.swift [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                // User tapped to place line of sight observer.
+                if lineOfSight == nil {
+                    // Create and set initial line of sight analysis with tapped scene point.
+                    lineOfSight = LocationLineOfSight(observerLocation: scenePoint!, targetLocation: scenePoint!)
+                    analysisOverlay.addAnalysis(lineOfSight!)
+                    shouldShowTargetInstruction = true
+                } else {
+                    // Update the observer location.
+                    lineOfSight?.observerLocation = scenePoint!
+                }
```
Comment (rolson) [inline Shared/Samples/Show line of sight between points/ShowLineOfSightBetweenPointsView.swift:58]: Here we have to assume that `scenePoint` is optional for a reason.  Therefore it's not safe to force unwrap it and could lead to crashes.

```suggestion
                guard let scenePoint else { return }
                if lineOfSight == nil {
                    // Create and set initial line of sight analysis with tapped scene point.
                    lineOfSight = LocationLineOfSight(observerLocation: scenePoint, targetLocation: scenePoint)
                    analysisOverlay.addAnalysis(lineOfSight!)
                    shouldShowTargetInstruction = true
                } else {
                    // Update the observer location.
                    lineOfSight?.observerLocation = scenePoint
                }
```

**Review by rolson — APPROVED** [review]

File: Shared/Samples/Show line of sight between points/ShowLineOfSightBetweenPointsView.swift [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                    analysisOverlay.addAnalysis(lineOfSight!)
+                    shouldShowTargetInstruction = true
+                } else {
+                    // Update the observer location.
+                    lineOfSight?.observerLocation = scenePoint
+                }
+            }
+            .onLongPressGesture {  _, scenePoint in
+                // Update the target location on long press.
+                lineOfSight?.targetLocation = scenePoint!
```
Comment (pgruenler) [inline Shared/Samples/Show line of sight between points/ShowLineOfSightBetweenPointsView.swift:63]: Would it make sense to have a guard for scenePoint here too?

**Review by pgruenler — COMMENTED** [review]
Looks good. Just a few comments.

**Review by rolson — APPROVED** [review]

**Review by pgruenler — APPROVED** [review]

File: Shared/Samples/Show line of sight between points/ShowLineOfSightBetweenPointsView.swift [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+import ArcGISToolkit
```
Comment (yo1995) [inline Shared/Samples/Show line of sight between points/ShowLineOfSightBetweenPointsView.swift:17]: This seems unneeded 👀

**Pull request merged**